os/mm/mm_heap/mm_calloc.c

```diff
@@ -82,13 +82,21 @@ FAR void *mm_calloc(FAR struct mm_heap_s *heap, size_t n, size_t elem_size)
 {
 	FAR void *ret = NULL;
 
-	if (n > 0 && elem_size > 0) {
+	if ((n == 0) || (elem_size == 0)) {
+		mdbg("Invalid Parameter, n : %u, elem_size : %u\n", n, elem_size);
+		return NULL;
+	}
+
+	if (n > (MMSIZE_MAX / elem_size)) {
+		mdbg("Parameter n(%u) should be smaller than %u, \
+			because multiplication of n and elem_size cannot overflow the size_t.\n", n, elem_size);
```

Comment:
Maybe second `%u` is `(MMSIZE_MAX / elem_size)`, not `elem_size`. Could you check it?
Because `n` could be bigger than `elem_size`, it' does not cause a problem.